src/context/abstractCast.ml

```diff
@@ -7,6 +7,7 @@ open Error
 
 let cast_stack = new_rec_stack()
```

Comment:
This one is difficult because the control flow goes deep into the call unification only to come back via `cast_or_unify_raise_ref` at some point. Piping a local context through all that would be insanity.

Fortunately, the relevant code should only matter for actual typing, which I don't plan to parallelize anyway.